mypy/fastparse.py

```diff
@@ -279,6 +279,16 @@ def parse(
         # Disable deprecation warnings about \u
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", category=DeprecationWarning)
+            if options.allow_commented_type_annotations:
+                lines = str(source).split("\n")
+                result = ""
+                for line in lines:
+                    if "# type:" not in str(line):
+                        result += str(line) + str("\n")
+                    else:
+                        line = line.split("#")[0] + "\n"
+                        result += line
+                    source = result
```

Comment:
I would really recommend introducing a change _after_ the "ast" phase:
https://github.com/python/mypy/blob/70d4bb27a2733e43958dc5046befdc16ed297a5c/mypy/fastparse.py#L334-L367

The code I've highlighted here handles a type comment that was found by AST. The specific lines 352-362 handle `# type: ignore`s, while lines 363-367 handle actual type annotations.